File: Data Structures & Algorithms/insert-new-interval/submission-6.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        i = 0
        while i < len(intervals) and intervals[i][0] < newInterval[0]: i += 1
        intervals.insert(i, newInterval)

        i = 0
        while i < len(intervals) - 1:
            while i+1 < len(intervals) and intervals[i][1] >= intervals[i+1][0]:
                intervals[i][1] = max(intervals[i][1], intervals[i+1][1])
                intervals.pop(i+1)
            i += 1
        
        return intervals



        

File: Data Structures & Algorithms/house-robber/submission-6.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        if (nums.size() == 1) {return nums[0];}
        int tmp;
        int a = nums[1];
        int b = nums[0];

        for (int i = 2;i < nums.size(); ++i) {
            tmp = a;
            a = max(a,b + nums[i]);
            b = max(tmp, b);
        }

        return max(a,b);


        

        
    }
};
